src/android/Dgram.java

```diff
@@ -27,9 +27,17 @@
     private DatagramSocket datagramSocket;
     private DatagramSocketListener datagramSocketListener;
 
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+
+        Log.d(TAG, "onDestroy: cleanup thread and socket");
+        closeSocket();
+    }
+
     @Override
     public boolean execute(
```

Comment:
pretty sure vscode just made a bunch of edits to strip out whitespace (I have that setting turned on).